libp2p/crypto/p2pSigner.go

```diff
@@ -54,7 +58,7 @@ func (signer *p2pSigner) Verify(payload []byte, pid core.PeerID, signature []byt
 }
 
 // SignUsingPrivateKey will sign the payload with provided private key bytes
-func (signer *p2pSigner) SignUsingPrivateKey(skBytes []byte, payload []byte) ([]byte, error) {
+func (signer *p2pSignerWrapper) SignUsingPrivateKey(skBytes []byte, payload []byte) ([]byte, error) {
```

Comment:
we still need this for the multi-key feature. Might be moved inside the netMessenger struct and this file can be removed
Also, the signing should be done according to a provided signer that will be the one provided on the netMessenger's signer

Reply:
i kept this wrapper at least for now, since there are some additional things over the single singer from crypto, like hashing over message, in order to align with the libp2p crypto implementation